spec/factories.rb

```diff
@@ -312,32 +327,32 @@ def r_str
   factory(:status, :parent => :status_message)
 
   factory :o_auth_application, class: Api::OpenidConnect::OAuthApplication do
-    client_name "Diaspora Test Client"
+    sequence(:client_name) {|n| "Diaspora Test Client #{n}#{r_str}" }
     redirect_uris %w(http://localhost:3000/)
   end
 
   factory :o_auth_application_with_image, class: Api::OpenidConnect::OAuthApplication do
-    client_name "Diaspora Test Client"
+    sequence(:client_name) {|n| "Diaspora Test Client #{n}#{r_str}" }
     redirect_uris %w(http://localhost:3000/)
     logo_uri "/assets/user/default.png"
   end
 
   factory :o_auth_application_with_ppid, class: Api::OpenidConnect::OAuthApplication do
-    client_name "Diaspora Test Client"
+    sequence(:client_name) {|n| "Diaspora Test Client #{n}#{r_str}" }
     redirect_uris %w(http://localhost:3000/)
     ppid true
     sector_identifier_uri "https://example.com/uri"
   end
 
   factory :o_auth_application_with_ppid_with_specific_id, class: Api::OpenidConnect::OAuthApplication do
-    client_name "Diaspora Test Client"
+    sequence(:client_name) {|n| "Diaspora Test Client #{n}#{r_str}" }
     redirect_uris %w(http://localhost:3000/)
     ppid true
     sector_identifier_uri "https://example.com/uri"
   end
 
   factory :o_auth_application_with_multiple_redirects, class: Api::OpenidConnect::OAuthApplication do
-    client_name "Diaspora Test Client"
+    sequence(:client_name) {|n| "Diaspora Test Client #{n}#{r_str}" }
```

Comment:
If the name is the same "Diaspora Test Client" for all objects, then with my patch [this](https://github.com/diaspora/diaspora/blob/develop/spec/controllers/api/openid_connect/clients_controller_spec.rb#L127) test fails. It happens because [this](https://github.com/diaspora/diaspora/pull/6783/commits/41049694dc870bf827513cf5446c01f8bd1be532#diff-8c42b18d6c734a3f806c8df8c01016b4R103) object creation depends on ```:o_auth_application``` factory and therefore we have an existing object with client_name "Diaspora Test Client" in the DB, while the test I've mentioned above makes search by the client_name and finds an unexpected object.

It seems that it is enough to make the name sequenced only for ```:o_auth_application``` factory, so I can revert name to the plain string for all the others.